docs/source/worker.rst

```diff
@@ -218,22 +225,182 @@ usage then the worker will start dumping unused data to disk, even if internal
+Dask — it's actually normal behavior for all processes on Linux and MacOS, and is a
+consequence of how the low-level memory allocator works (see below for details).
+
+Because Dask makes decisions (spill-to-disk, pause, terminate,
+:meth:`~distributed.Client.rebalance`) based on the worker's memory usage as reported by
+the OS, and is unaware of how much of this memory is actually in use versus empty and
+"hoarded", it can overestimate — sometimes significantly — how much memory the process
+is using and think the worker is running out of memory when in fact it isn't.
+
+More in detail: both the Linux and MacOS memory allocators try to avoid performing a
+`brk`_ kernel call every time the application calls `free`_ by implementing a user-space
+memory management system. Upon `free`_, memory can remain allocated in user space and
+potentially reusable by the next `malloc`_ - which in turn won't require a kernel call
+either. This is generally very desirable for C/C++ applications which have no memory
+allocator of their own, as it can drastically boost performance at the cost of a larger
+memory footprint. CPython however adds its own memory allocator on top, which reduces
+the need for this additional abstraction (with caveats).
+
+There are steps you can take to alleviate situations where worker memory is not released
+back to the OS. These steps are discussed in the following sections.
+
+Manually trim memory
+~~~~~~~~~~~~~~~~~~~~
+*Linux workers only*
+
+It is possible to forcefully release allocated but unutilized memory as follows:
+
+.. code-block:: python
+
+    import ctypes
+
+    def trim_memory() -> int:
+        libc = ctypes.CDLL("libc.so.6")
+        return libc.malloc_trim(0)
+
+    client.run(trim_memory)
+
+This should be only used as a one-off debugging experiment. Watch the dashboard while
+running the above code. If unmanaged worker memory (on the "Bytes stored" plot)
+decreases significantly after calling ``client.run(trim_memory)``, then move on to the
+next section. Otherwise, you likely do have a memory leak.
+
+Note that you should only run this `malloc_trim`_ if you are using the default glibc
+memory allocator. When using a custom allocator such as `jemalloc`_ (see below), this
+could cause unexpected behavior including segfaults. (If you don't know what this means,
+you're probably using the default glibc allocator and are safe to run this).
+
+Automatically trim memory
+~~~~~~~~~~~~~~~~~~~~~~~~~
+*Linux workers only*
+
+To aggressively and automatically trim the memory in a production environment, you
+should instead set the environment variable ``MALLOC_TRIM_THRESHOLD_`` (note the final
+underscore) to 0 or a low number; see the `mallopt`_ man page for details. Reducing
+this value will increase the number of syscalls, and as a consequence may degrade
+performance. **The variable must be set before starting the ``dask-worker`` process.**
+
+jemalloc
+~~~~~~~~
```

Comment:
A further alternative nowadays is `mimalloc`. It works in this case just as `jemalloc` but is available on Windows, too.